content/WebAuthn/Concepts/Using_WebAuthn_for_Signing.adoc

```diff
@@ -0,0 +1,25 @@
+
+Further, the link:https://www.w3.org/TR/webauthn-2/#enum-userVerificationRequirement[User Verification] behavior is also defined by the MakeCredential operation, allowing a relying party to define if the created credential will require a PIN or biometric supplied by the user to access. This can allow for securing keypairs used for signing to only valid users of the WebAuthn Authenticator. In addition, it is also recommneded link:https://www.w3.org/TR/webauthn-2/#sctn-attestation[Attestation] is enabled during the registration event to verify the authenticity of the WebAuthn authenticator being used. Other fields in the MakeCredential can be provided with other identifying information or left to default values. However, it is important to note the orginID should match the endpoint the user is provided to access the signing service, as is the case in a standard WebAuthn service.
+
+With the MakeCredential operation creating a keypair with the algorithm defined, the link:https://www.w3.org/TR/webauthn-2/#sctn-op-get-assertion[GetAssertion] operation is used to actually sign the data. The GetAssertion flow also supplies a challenge from the relying party to the authenticator to be signed as part of the link:https://www.w3.org/TR/webauthn-2/#sctn-verifying-assertion[Authentication] ceremony. This challenge can be replaced with a hash of the document to be signed, creating an association with the document or data set to be signed. The method in which this hash is generated should be determined by the capabilities of the environment and security requirements under which the application is run. 
+
+No modifications are required to this operation to support the signature, but both the link:https://www.w3.org/TR/webauthn-1/#dictdef-collectedclientdata[client data] and link:https://www.w3.org/TR/webauthn-1/#sec-authenticator-data[authenticator data] will need to be recorded along with the returned signature. This is because the signature returned by WebAuthn is for an object comprising both the client data (including the hash string) and authenticator data, and both sets of data will be required to validate the signature in the future. To assist with the, the PublicKeyCredential returned will include the client data (as a JSON object), the authenticator data and the signature. Simply archiving everything returned will allow for ease of signature validation. In a pinch, the client data does not absolutely need to be retained as it can be reconstructed as long as the domain is known and the relying party remains operational, but for ease of access and export, there are benefits for simply storing it. Other metadata, such as the timestamp of the signing event, may be useful to include as well.
```

Comment:
The client data cannot be reliably reconstructed, it MUST be stored. Not only are there many different valid JSON serializations of it, but it also includes new information from the client that the RP cannot know independently, like the `crossOrigin` and `tokenBinding` properties (the `origin` property also MUST be taken from the client for proper security, but I'll concede that one may assume that the database entry was validated before storage, so `origin` is known implicitly for stored signatures). There may also be new properties added in the future, and Chrome already adds a dummy extra value sometimes at random.

```suggestion
No modifications are required to this operation to support the signature, but both the link:https://www.w3.org/TR/webauthn-1/#dictdef-collectedclientdata[client data] and link:https://www.w3.org/TR/webauthn-1/#sec-authenticator-data[authenticator data] will need to be recorded along with the returned signature. This is because the signature returned by WebAuthn is for an object comprising both the client data (including the hash string) and authenticator data, and both sets of data will be required to validate the signature in the future. To assist with the, the PublicKeyCredential returned will include the client data (as a JSON object), the authenticator data and the signature. Simply archiving everything returned will allow for ease of signature validation. Other metadata, such as the timestamp of the signing event, may be useful to include as well.
```

Reply:
Removed reference to re-building client data. Current suggestions is to archive everything.